### PR TITLE
[eclipse/xtext-eclipse#520] Adjust to latest API

### PR DESCRIPTION
**Pull request by kthoms**
The code mining provider base class has been renamed and only offers one
abstract method to implement by clients.
Also some minor cleanups.

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend [hunk 1]
```diff
@@ -90,27 +92,19 @@ class CodeMiningFragment extends AbstractStubGeneratingFragment {
...
-					//TODO: implement me
-					//use acceptor.accept(super.createNewLineHeaderCodeMining(...)) to add a new code mining to the final list
-					
-					//example:
-					//acceptor.accept(createNewLineHeaderCodeMining(1, document, "Header annotation"))
-				}
-				
-				override void createLineContentCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator, IAcceptor<ICodeMining> acceptor)  throws BadLocationException {
+			class �codeMiningProviderClass.lastSegment� extends AbstractXtextCodeMiningProvider {
+				override void createCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator,
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: @szarnekow  @kthoms isnt it convention to use "typeRefs" or <<XtextResource>> at such places?

Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: (instead of using the imports?)

Comment (szarnekow) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: True, that would be cleaner.

Comment (kthoms) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: done.

**Pull request merged**